File: umbrella-insurance-react-ui/src/models/users/passwords/v1/Password.ts
import { User } from "../../v1/User";

export class Password {
    id?: number;
    user?: User;
    salt?: string;
    password?: string;
    hashedPassword?: string;
    createdDateTime?: string;

    constructor(obj?:any) {
        if(obj) {
            if(obj?.id) {
                this.id = obj.id;
            }
            if(obj?.user) {
                this.user = new User(obj.user);
            }
            if(obj?.salt) {
                this.salt = obj.salt;
            }
            if(obj?.password) {
                this.password = obj.password;
            }
            if(obj?.hashedPassword) {
                this.hashedPassword = obj.hashedPassword;
            }
            if(obj?.createdDateTime) {
                this.createdDateTime = obj.createdDateTime;
            }
        }
    }
}

File: umbrella-insurance-core/src/main/java/com/umbrella/insurance/core/models/entities/Password.java
package com.umbrella.insurance.core.models.entities;

import com.umbrella.insurance.core.deserializers.TimestampDeserializer;
import com.umbrella.insurance.core.serializers.TimestampSerializer;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import java.sql.Timestamp;

@Getter
@Setter
@Entity
@Table(name = "passwords", schema = "public")
public class Password {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "password_id", nullable = false)
    private Long id;

    @Column(name = "salt", length = Integer.MAX_VALUE)
    private String salt;

    @Column(name = "hashed_password", length = Integer.MAX_VALUE)
    private String hashedPassword;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "user_id")
    private User user;

    @JsonSerialize(using = TimestampSerializer.class)
    @JsonDeserialize(using = TimestampDeserializer.class)
    @Column(name = "created_date_time")
    private Timestamp createdDateTime;

}